core/components/header/index.tsx

```diff
@@ -95,15 +92,34 @@ const getCartCount = async () => {
   return response.data.site.cart.lineItems.totalQuantity;
 };
 
+const getCurrencies = async () => {
+  const data = await getLayoutData();
+
+  if (!data.currencies.edges) {
+    return [];
+  }
+
+  const currencies = data.currencies.edges.map(({ node }) => ({
+    code: node.code,
+  }));
+
+  return currencies;
+};
+
 export const Header = async () => {
   const t = await getTranslations('Components.Header');
   const locale = await getLocale();
+  const currencyCode = await getPreferredCurrencyCode();
 
   const locales = routing.locales.map((enabledLocales) => ({
     id: enabledLocales,
     label: enabledLocales.toLocaleUpperCase(),
   }));
 
+  const currencies = await getCurrencies();
+  // todo handle default currency properly once added to API
```

Comment:
🍹 Easier for searchability and works ootb with the `TODO Highlight` plugin
```suggestion
  // TODO: handle default currency properly once added to API
```
![Screenshot 2025-01-16 at 10 38 19](https://github.com/user-attachments/assets/1ab883a9-8e60-4c1d-b185-d5ab60c2a812)